### PR TITLE
Replaces hardcoded gpu_type and gpu_mem

### PR DESCRIPTION
**Pull request by orangetin**
Replaces hardcoded values for `gpu_type` and `gpu_mem` to info from torch.cuda

For example, `gpu_type="NVIDIA A100 80GB PCIe"` and `gpu_mem="82998656"`.

### CODE REVIEW

File: serving_local_nlp_model.py [hunk 1]
```diff
@@ -419,8 +419,8 @@ def _run_inference(self):
         "group_name": args.group_name,
         "max_batch_size": args.max_batch_size,
         "gpu_num":1,
-        "gpu_type":"RTX 3090",
-        "gpu_mem":2400000,
+        "gpu_type": torch.cuda.get_device_name(device="cuda:0"),
+        "gpu_mem": str(int(torch.cuda.mem_get_info()[-1]/1024)),
```
Comment (csris) [inline serving_local_nlp_model.py:423]: Let's do this only if cuda is available.

```suggestion
            "gpu_type": (
                torch.cuda.get_device_name(0) if torch.cuda.is_available() else None
            ),
            "gpu_mem": (
                torch.cuda.get_device_properties(0).total_memory
                if torch.cuda.is_available()
                else None
            ),
```

Additionally, I'm not sure how to handle this but if we're on a system with different gpu types, this only reports the first one. Probably not worth changing the types right now, but something to note.

Comment (csris) [reply serving_local_nlp_model.py]: Also don't divide by 1024. Let's retain the precision and leave it to the user-facing code to decide how to display it.

Comment (orangetin) [reply serving_local_nlp_model.py]: > Additionally, I'm not sure how to handle this but if we're on a system with different gpu types, this only reports the first one. Probably not worth changing the types right now, but something to note.

Fixed this.

I'll make the change for if cuda is available too

**Review by csris — APPROVED** [review]
Great. Thanks!

**Comment by orangetin** [comment]
@csris It now checks if cuda is available, and if it is, uses `torch.cuda.current_device` as the default device rather than the first.

**Pull request merged**